pkg/webservice/handlers.go

```diff
@@ -241,23 +241,24 @@ func GetApplicationHistory(w http.ResponseWriter, r *http.Request) {
 	writeHeaders(w)
 
 	if imHistory != nil {
-		var result []*dao.ApplicationHistoryDAOInfo
-		records := imHistory.GetRecords()
-		for _, record := range records {
-			if record == nil {
-				break
-			}
-			element := &dao.ApplicationHistoryDAOInfo{
-				Timestamp:         record.Timestamp.UnixNano(),
-				TotalApplications: strconv.Itoa(record.TotalApplications),
-			}
-			result = append(result, element)
+		http.Error(w, "Internal metrics collection is not enabled.", http.StatusInternalServerError)
+		return
+	}
+	var result []*dao.ApplicationHistoryDAOInfo
+	records := imHistory.GetRecords()
+	for _, record := range records {
+		if record == nil {
+			continue
```

Comment:
We still should break here: internal filling of the array stops when we have run out of stats to collect. If we cannot convert we set it to -1. So the only way the record can be nil is if we have never written to it and that means all follow up values are nil.

Reply:
As I wrote in my comment [above](https://github.com/apache/incubator-yunikorn-core/pull/125#discussion_r404605856) in more detail, the array returned by the `GetRecords()` may have `nil` values **at the beginning**, but have non-nil values later.